oaknational/Oak-Web-Application
# CTC-27: teacher preview lesson page

## Description

Music year: 1963

#### Creates a preview lesson page hidden behind a beta route

- Updates sdk
- Adds `teacherPreviewLesson` query and test file
- Creates page route `/teachers/beta/lessons/[lessonSlug] only accessible with new / unpublished data
- Adds a test file for new page

#### N.B. If data is incomplete it may cause the page to throw a 404, some data has been hard coded as we are utilising browse fixture data

#### Teacher beta redirects implemented with this [PR #ENG-1028 ](https://github.com/oaknational/Cloud-Config/pull/330)

## Issue(s)

Fixes #CTC-27

ACs:
- [x]  I can see a `new` lesson in OWA at teachers/beta/lessons/[lessonSlug]
- [x]  New lessons are not accessible outside of Oak
- [x]  New lessons use the fallback subject icon
- [ ]  I can preview new lessons from the CAT

#### Changes to preview new lessons made in CAT, currently hidden until this PR is merged

## How to test

1. Go to https://deploy-preview-2952--oak-web-application.netlify.thenational.academy/teachers/beta/lessons/running-as-a-team or use the slug of any new lesson
2. You should see a lesson overview page rendering the data of the new lesson 

## Screenshots

How it should look:

<img width="1339" alt="Screenshot 2024-11-04 at 16 35 36" src="https://github.com/user-attachments/assets/d140da69-a8af-490d-a94c-97fea9acb917">

## Checklist
